frontend/src/lib/v2/WorkflowUtils.ts

```diff
@@ -21,6 +21,18 @@ import { Workflow } from 'src/third_party/mlmd/argo_template';
 
 // This key is used to retrieve the platform-agnostic pipeline definition
 export const PIPELINE_SPEC_TEMPLATE_KEY = 'pipeline_spec';
+export const PLATFORM_SPEC_TEMPLATE_KEY = 'platform_spec';
+
+export function getPipelineDefFromYaml(template: string) {
+  // If pipeline_spec exists in the return value of safeload,
+  // which means the original yaml contains platform_spec,
+  // then the PipelineSpec(IR) is stored in 'pipeline_spec' field.
+  return jsyaml.safeLoad(template)[PIPELINE_SPEC_TEMPLATE_KEY] ?? jsyaml.safeLoad(template);
+}
+
+export function getPlatformDefFromYaml(template: string) {
+  return jsyaml.safeLoad(template)[PLATFORM_SPEC_TEMPLATE_KEY];
+}
```

Comment:
We might need to add some tests to validate return values `when kubernetes-platform exists` and `when no platform spec exists` of this exported function.